kore/src/Kore/Exec.hs

```diff
@@ -220,15 +212,36 @@ exec
     -> smt (TermLike VariableName)
 exec breadthLimit verifiedModule strategy initialTerm =
     evalSimplifier verifiedModule' $ do
-        execution <- execute breadthLimit verifiedModule' strategy initialTerm
-        let
-            Execution { executionGraph } = execution
-            finalConfig = pickLongest executionGraph
-            finalTerm =
-                forceSort patternSort
-                $ Pattern.toTermLike finalConfig
+        initialized <- initialize verifiedModule
+        let Initialized { rewriteRules } = initialized
+        finalConfig <-
+            getFinalConfigOf $ do
```

Comment:
```suggestion
        (_, finalConfig) <-
            Logic.observeT $ do
```

Reply:
[`observeT`](https://hackage.haskell.org/package/logict-0.7.0.2/docs/Control-Monad-Logic.html#v:observeT) invokes `fail` if there is not at least one result, but I can change this to be somewhat more idiomatic.